analytic_tag_dimension/models/analytic_dimension_line.py

```diff
@@ -10,6 +11,32 @@ class AnalyticDimensionLine(models.AbstractModel):
     _description = "Analytic Dimension Line"
     _analytic_tag_field_name = "analytic_tag_ids"
 
+    @api.model
+    def get_model_names(self):
+        return [
+            "account.move.line",
+            "account.analytic.line",
+            "account.invoice.report",
+        ]
+
+    def _reflect(self):
+        super()._reflect()
+        if (
+            self._inherit != self._name
+            and self._inherit in self.get_model_names()
+            and self.env.context.get("check_fields", True)
+        ):
+            fields_inherit = self.env["ir.model.fields"].search(
+                [
+                    ("model_id.model", "=", self._inherit),
+                    ("name", "not in", self.fields_get_keys()),
+                ]
+            )
+            if fields_inherit:
+                model = self.env["ir.model"].search([("model", "=", self._name)])
+            for field in fields_inherit:
+                field.with_context(check_fields=False).copy({"model_id": model.id})
```

Comment:
What this exactly does? In the ORM, this is creating a copy of the ir.model.fields record, so not sure this is correct.

Reply:
It's necessary and create `ir.model.field` records lost.
When dimension record is created, auto-create "custom fields" in some models according to: https://github.com/OCA/account-analytic/blob/13.0/analytic_tag_dimension/models/account_analytic_dimension.py#L38-L59 but these process have side-efects but now work fine.

Example:
- `account.move.line` (inherit from `analytic.dimension.line`) > Before, fields are auto-created in these model.
- `custom.wizard.model` (inherit from `account.move.line`) > Now, fields are auto-created in these model too.